src/packet/public_key.js

```diff
@@ -100,8 +104,13 @@ PublicKey.prototype.read = function (bytes) {
     var p = 0;
 
     for (var i = 0; i < mpicount && p < bmpi.length; i++) {
-
-      this.mpi[i] = new type_mpi();
+      if ((this.algorithm === 'ecdsa' || this.algorithm === 'ecdh') && i === 0) {
```

Comment:
Here is the issue related to the assumption that key packets only contain MPI. In order to support Ed25519 also a similar change have to be made.